src/Swarm/Language/Pretty.hs

```diff
@@ -299,37 +299,42 @@ prettyTypeErr code (CTE l tcStack te) =
   showLoc (r, c) = pretty r <> ":" <> pretty c
 
 instance PrettyPrec TypeErr where
-  prettyPrec _ (UnifyErr ty1 ty2) =
-    "Can't unify" <+> ppr ty1 <+> "and" <+> ppr ty2
-  prettyPrec _ (Mismatch Nothing (getJoin -> (ty1, ty2))) =
-    "Type mismatch: expected" <+> ppr ty1 <> ", but got" <+> ppr ty2
-  prettyPrec _ (Mismatch (Just t) (getJoin -> (ty1, ty2))) =
-    nest 2 . vcat $
-      [ "Type mismatch:"
-      , "From context, expected" <+> bquote (ppr t) <+> "to" <+> typeDescription Expected ty1 <> ","
-      , "but it" <+> typeDescription Actual ty2
-      ]
-  prettyPrec _ (LambdaArgMismatch (getJoin -> (ty1, ty2))) =
-    "Lambda argument has type annotation" <+> bquote (ppr ty2) <> ", but expected argument type" <+> bquote (ppr ty1)
-  prettyPrec _ (FieldsMismatch (getJoin -> (expFs, actFs))) = fieldMismatchMsg expFs actFs
-  prettyPrec _ (EscapedSkolem x) =
-    "Skolem variable" <+> pretty x <+> "would escape its scope"
-  prettyPrec _ (UnboundVar x) =
-    "Unbound variable" <+> pretty x
-  prettyPrec _ (Infinite x uty) =
-    "Infinite type:" <+> ppr x <+> "=" <+> ppr uty
-  prettyPrec _ (DefNotTopLevel t) =
-    "Definitions may only be at the top level:" <+> ppr t
-  prettyPrec _ (CantInfer t) =
-    "Couldn't infer the type of term (this shouldn't happen; please report this as a bug!):" <+> ppr t
-  prettyPrec _ (CantInferProj t) =
-    "Can't infer the type of a record projection:" <+> ppr t
-  prettyPrec _ (UnknownProj x t) =
-    "Record does not have a field with name" <+> pretty x <> ":" <+> ppr t
-  prettyPrec _ (InvalidAtomic reason t) =
-    "Invalid atomic block:" <+> ppr reason <> ":" <+> ppr t
-  prettyPrec _ Impredicative =
-    "Unconstrained unification type variables encountered, likely due to an impredicative type. This is a known bug; for more information see https://github.com/swarm-game/swarm/issues/351 ."
+  prettyPrec _ = \case
+    (UnifyErr ty1 ty2) ->
```

Comment:
Aren't all these parentheses around the different cases redundant?

Reply:
They are, I just did not get them with my initial regex and then I was lazy to do it. 😅